### DISCUSSION TITLE
Best practice for referring to external terms and ontologies

### DISCUSSION THREAD

**Discussion by sneumann** [BioSchemas/specifications#618] [General]
Hi, people generally like the idea of using controlled terminology in properties, and we have wonderful ontologies in OBO and elsewhere. However, multiple approaches with their own pros/cons exist to capture and link to the external terminologies. This is the place to collect use cases, examples and discuss the approaches.

**Comment by gtsueng** [comment]
Some related issues to help jump-start the discussion: 
* https://github.com/BioSchemas/specifications/issues/594
* https://github.com/schemaorg/schemaorg/issues/2734

**Reply by ljgarcia** [reply]
Adding DefinedTerm to measurementTechnique as possible expected type could have unintended consequences as it is not part of schema.org Will it break, e.g., Google Dataset validation?

**Comment by marco-brandizi** [comment]
This is also related to attaching ontology qualifiers to `PropertyValue` instances, see schemaorg/schemaorg#2986 and #538. Within the Agrischemas and MIAPPE-2-bioschemas projects, we're currently using `dc:type` to link ontology terms, see [here](https://github.com/Rothamsted/agri-schemas/tree/master/doc/miappe-use-case/dataset-arabidopsis#valuetype-pairs).

**Reply by ljgarcia** [reply]
But this would only work if PropertyValue is allowed within the range of the property, right?

**Comment by ljgarcia** [comment]
@sneumann as this discussion goes beyond the Bioschemas community, i.e., it would require changes in schema.org, I created an issue [there](https://github.com/schemaorg/schemaorg/issues/3250). Hopefully we can make a change.

**Comment by ljgarcia** [comment]
Here is the list of types and their properties where DefinedTerm could be added to the expected range
* FormalParameter: encodingFormat (it already suggests using EDAM which is a DefinedTerm from schema.org perspective)
* ComputationalTool: applicationCategory and applicationSubCategory (DefinedTerm in addition to Text and URL, EDAM is already suggested), featureList (in addition to Text and URL), keywords (DefinedTerm exists in schema.org, should not be removed), operatingSystem, programmingLanguage
* ComputationalWorkflow: encodingFormat
* CourseInstance: courseMode, eventStatus
* DataCatalog: keywords, encodingFormat
* Dataset: measurementTechnique, variableMeasured
